### PR TITLE
[Clang] Rename and enable boolean get, set, create and undef for sme2

### PR DESCRIPTION
**Pull request by SamTebbs33**
This patch renames the get, set, create and undef functions that deal with tuples of booleans to match the ACLE at
https://github.com/ARM-software/acle/pull/257/files . It also enables them for SME2.

### CODE REVIEW

File: clang/include/clang/Basic/arm_sve.td [hunk 1]
```diff
@@ -1321,12 +1321,17 @@ def SVSET_3_BF16 : SInst<"svset3[_{d}]", "33id", "b", MergeNone, "", [IsTupleSet
 def SVSET_4_BF16 : SInst<"svset4[_{d}]", "44id", "b", MergeNone, "", [IsTupleSet], [ImmCheck<1, ImmCheck0_3>]>;
 }
 
-let TargetGuard = "sve2p1" in {
-  def SVGET_2_B : SInst<"svget2[_{d}]", "d2i", "Pc", MergeNone, "", [IsTupleGet], [ImmCheck<1, ImmCheck0_1>]>;
-  def SVGET_4_B : SInst<"svget4[_{d}]", "d4i", "Pc", MergeNone, "", [IsTupleGet], [ImmCheck<1, ImmCheck0_3>]>;
+let TargetGuard = "sve2p1|sme2" in {
+  def SVGET_2_B : SInst<"svget2_b", "d2i", "Pc", MergeNone, "", [IsTupleGet], [ImmCheck<1, ImmCheck0_1>]>;
+  def SVGET_4_B : SInst<"svget4_b", "d4i", "Pc", MergeNone, "", [IsTupleGet], [ImmCheck<1, ImmCheck0_3>]>;
```
Comment (kmclaughlin-arm) [inline clang/include/clang/Basic/arm_sve.td:1326]: Does the `_b` still need to be in square brackets? I think this part of the name is optional for get, set & create (but not undef).

Comment (SamTebbs33) [reply clang/include/clang/Basic/arm_sve.td]: Thanks Kerry, I've made _b optional and also added IsStreamingCompatible to the get and set builtins.

File: clang/test/CodeGen/aarch64-sve2p1-intrinsics/acle_sve2p1_undef_bool.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 4
+// RUN: %clang_cc1 -triple aarch64-none-linux-gnu -target-feature +sve2p1 -S -O2 -Werror -Wall -emit-llvm -o - %s | opt -S -passes=mem2reg,tailcallelim | FileCheck %s
+// RUN: %clang_cc1 -triple aarch64-none-linux-gnu -target-feature +sve2p1 -S -O2 -Werror -Wall -emit-llvm -o - -x c++ %s | opt -S -passes=mem2reg,tailcallelim | FileCheck %s -check-prefix=CPP-CHECK
+// RUN: %clang_cc1 -triple aarch64-none-linux-gnu -target-feature +sme2 -S -O2 -Werror -Wall -emit-llvm -o - %s | opt -S -passes=mem2reg,tailcallelim | FileCheck %s
+// RUN: %clang_cc1 -triple aarch64-none-linux-gnu -target-feature +sme2 -S -O2 -Werror -Wall -emit-llvm -o - -x c++ %s | opt -S -passes=mem2reg,tailcallelim | FileCheck %s -check-prefix=CPP-CHECK
```
Comment (kmclaughlin-arm) [inline clang/test/CodeGen/aarch64-sve2p1-intrinsics/acle_sve2p1_undef_bool.c:5]: Please can you also add a line to test these builtins end to end? For example:
`RUN: %clang_cc1 -triple aarch64-none-linux-gnu -target-feature +sme2 -S -disable-O0-optnone -Werror -Wall -o /dev/null %s`

Comment (SamTebbs33) [reply clang/test/CodeGen/aarch64-sve2p1-intrinsics/acle_sve2p1_undef_bool.c:5]: Ah thanks for spotting that, added now.

**Review by kmclaughlin-arm — APPROVED** [review]

**Pull request merged**